### DISCUSSION TITLE
sql component can not replace pgsql schema or tablename parameter

### DISCUSSION THREAD

**Discussion by hangliu1994** [apache/dolphinscheduler#9144] [Q&A]
When i use a parameter  in sql component to replace pgsql schema,It will indicate that it cannot be replaced.

The following image is my script：
  
![image](https://user-images.githubusercontent.com/59187866/159831140-012ec3ec-de1a-4aa4-9ebe-0489c5adecec.png)

The following image is log:
![image](https://user-images.githubusercontent.com/59187866/159831047-8c4e90a1-02cd-4f03-a0ac-fcd0c03cc04d.png)

How to replace pgsql schema param? Please help me

**Comment by hangliu1994** [comment]
@710700

**Comment by Tandoy** [comment]
Maybe try this ${schema}

SELECT count(1) from ${schema}.wh$wh_official_account_follow_cnt_bd;

![企业微信截图_20220325095237](https://user-images.githubusercontent.com/56899730/160039097-649cd163-396d-40b7-bec0-f3d1c9ee91d4.png)

![企业微信截图_20220325095257](https://user-images.githubusercontent.com/56899730/160039106-16e48302-b986-4ae9-a074-12ef96005453.png)

**Reply by hangliu1994** [reply]
Please tell me which datasource you choose? 
When I use mysql ,it is effective, but postgresql is not effective.

**Reply by hangliu1994** [reply]
My dolphinscheduler  release is 2.0.3

**Reply by Tandoy** [reply]
> My dolphinscheduler release is 2.0.3

The data source is also postgresql, but the version is 1.3.6, I will try the new version later for experiments

**Reply by hangliu1994** [reply]
> > My dolphinscheduler release is 2.0.3
> 
> The data source is also postgresql, but the version is 1.3.6, I will try the new version later for experiments

Thank you for your help
